websites/R/Rate Your Music/presence.ts

```diff
@@ -0,0 +1,286 @@
+				presenceData.state = genreName;
+			}
+		}
+	}
+
+	//artist page
+	if (pathname.startsWith("/artist/")) {
+		const artistNameElement = document.querySelector("h1.artist_name_hdr");
+		if (artistNameElement) {
+			const artistName = artistNameElement.textContent?.trim();
+			if (artistName) {
+				presenceData.details = "Viewing Artist:";
+				presenceData.state = artistName;
+			}
+		}
+	}
+
+	//user page
+	if (pathname.startsWith("/~")) {
+		const usernameElement = document.querySelector("#profilename");
+		if (usernameElement) {
+			const username = usernameElement.textContent?.trim();
+			if (username) {
+				presenceData.details = "Viewing User:";
+				presenceData.state = username;
+			}
+		}
+	}
+
+	//list page
+	if (pathname.startsWith("/list/")) {
+		const listTitleElement = document.querySelector(
+				"h1[style='font-size:2.1em']"
+			),
+			listOwner = pathname.split("/")[2];
+		if (listTitleElement) {
+			const listTitle = listTitleElement.textContent?.trim();
+			if (listTitle && listOwner) {
+				presenceData.details = "Viewing List:";
+				presenceData.state = `${listTitle} by ${listOwner}`;
+			}
+		}
+	}
+
+	//voting on descriptors page
+	if (pathname.startsWith("/rdescriptor/set")) {
+		const albumElement = document.querySelector("a.album"),
+			artistElement = document.querySelector("a.artist");
+		if (albumElement && artistElement) {
+			presenceData.details = "Voting on Descriptors";
+			presenceData.state = `${albumElement.textContent?.trim()} by ${artistElement.textContent?.trim()}`;
+		}
+	}
+
+	//editing list page
+	if (pathname.startsWith("/lists/edit") && document.querySelector("span"))
+		presenceData.details = "Editing List:";
+
+	// search page
+	if (pathname.startsWith("/search")) {
+		presenceData.details = "Searching for:";
+		presenceData.state = new URLSearchParams(window.location.search).get(
+			"searchterm"
+		);
+	}
+
+	//release page
+	if (pathname.startsWith("/release/")) {
+		const albumTitleElement = document.querySelector(".album_title");
+		if (
+			albumTitleElement &&
+			document.querySelector(".album_artist_small a.artist")
+		) {
+			presenceData.details = "Viewing Release:";
+			presenceData.state = `${albumTitleElement.textContent?.trim()}`;
+		}
+	}
+
+	//collection page
+	if (pathname.startsWith("/collection/")) {
+		presenceData.details = "Viewing Collection:";
+		presenceData.state = `${pathname.split("/")[2]}`;
+	}
+
+	// compatibility page
+	if (pathname.startsWith("/find_similar_users")) {
+		presenceData.details = "Viewing Compatibility:";
+		presenceData.state = new URLSearchParams(window.location.search).get(
+			"user"
+		);
+	}
+
+	for (const [path, data] of Object.entries(pages)) {
+		if (pathname === path) {
+			presenceData = { ...presenceData, ...data };
+			break;
+		}
+	}
+
+	new Presence({
+		clientId: "1213365863470866462",
+	}).setActivity(presenceData);
+}
+
+updatePresence();
+
+window.addEventListener("popstate", updatePresence);
+
+setInterval(updatePresence, 3000);
```

Comment:
Lines 277-279 may cause performance issues while browsing this site for some time.
To resolve this, create a single instance at the top of the script, and/or use the provided `UpdateData` event listener to call your `updatePresences` method.